dateparser/languages/language.py

```diff
@@ -1,29 +1,46 @@
 # -*- coding: utf-8 -*-
+import hashlib
 import re
 from itertools import chain
 
 from dateutil import parser
 
 from dateparser.timezone_parser import pop_tz_offset_from_string
-from dateparser.utils import wrap_replacement_for_regex
+from dateparser.utils import wrap_replacement_for_regex, registry
 
 from .dictionary import Dictionary, ALWAYS_KEEP_TOKENS
 from .validation import LanguageValidator
 
 
+@registry
 class Language(object):
+
     _dictionary = None
     _splitters = None
     _wordchars = None
 
-    def __init__(self, shortname, language_info):
+    @classmethod
+    def get_key(cls, *args, **kwargs):
+        key_template = '%(shortname)s-%(skip_tokens)s'
```

Comment:
I think we either need to describe how to build this method in docs, or better yet, to find a way to refactor this, so `Language` class was not registry itself, but better yet pass settings object to `translate` to `_join` to `_get_splitters` and store cache of splitters in language instead. What do you think.